creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -0,0 +1,125 @@
+const moment = require('moment')
+
+const redisClient = require('../redis')
+const { logger: genericLogger } = require('../logging')
+
+const SYNC_STATES = Object.freeze({
+  success: 'success',
+  fail: 'fail'
+})
+
+// Make key expire in 7 days in seconds
+const EXPIRATION = 7 /* days */ * 24 /* hr */ * 60 /* min */ * 60 /* s */
+
+/**
+ * Class intended to:
+ * - Record the number of the current day's successful and failed sync attempts
+ * - Record the timestamp of the most recent successful and failed sync
+ */
+
+// Note: When recording a sync 'success' or 'fail', it implies that the secondary was able to respectfully
+// successfully or unsuccessfully take the primary's exported data, and update its own state
+
+class SyncHistoryAggregator {
+  static async recordSyncSuccess (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.success,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncFail (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.fail,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncData ({ state, timeOfEvent, logContext }) {
+    const logger = genericLogger.child(logContext)
+
+    try {
+      // Update aggregate sync data
+      const aggregateSyncKeys = SyncHistoryAggregator.getAggregateSyncKeys()
+
+      // Get the TTL if the key exists for the number of `state` syncs, and update the value by 1
+      const aggregateSyncKeyStateTTL = await this.getKeyTTL(aggregateSyncKeys[state])
```

Comment:
what if `aggregateSyncKeys[state]` does note exist if e.g. `state` is invalid. maybe worth doing some param validation there but also probably not worth lol since this is all internal - just flagging :)

Reply:
also given repeated access prob should just store `aggregateSyncKeys[state]` in var

Reply:
I added a check condition to make sure that the state passed in is proper. also I dont know if storing `aggregateSyncKeys[state]` in a var will make a huge difference in performance or readability. i dont feel strongly but because I dont see much value with the change, I will leave it here unless you feel strongly

Reply:
thanks for adding the param validation

to your point about storing `aggregateSyncKeys[state]` in var, it will make no difference in performance and a very minor one in readability

can leave it